job-server/src/main/scala/spark/jobserver/AkkaClusterSupervisorActor.scala

```diff
@@ -152,8 +155,14 @@ class AkkaClusterSupervisorActor(daoActor: ActorRef) extends InstrumentedActor {
         logger.info("Shutting down context {}", name)
         val contextActorRef = contexts(name)._1
         cluster.down(contextActorRef.path.address)
-        contextActorRef ! PoisonPill
-        sender ! ContextStopped
+        try {
+          val stoppedCtx = gracefulStop(contexts(name)._1, 2 seconds)
```

Comment:
Good point. Should I add a new config option `jobserver.context-deletion-timeout` ?